src/Transition/VCompareGlobalRoutines.f90

```diff
@@ -749,14 +790,27 @@ SUBROUTINE CheckSpecialObjects(DifUnit,ObjectName,CurArgs,OutArgs,FieldNames,Fie
           CALL WriteOutIDFLinesAsSingleLine(DifUnit,ObjectName,CurArgs,OutArgs,FieldNames,FieldUnits)
 
         CASE('BUILDINGSURFACE:DETAILED')
-          CALL WriteOutPartialIDFLines(DifUnit,ObjectName,10,OutArgs,FieldNames,FieldUnits)
           IF (MakeUPPERCase(OutArgs(10)) == 'AUTOCALCULATE') THEN
             NVert=(CurArgs-10)/3
           ELSEIF (OutArgs(10) == '') THEN
             NVert=(CurArgs-10)/3
           ELSE
-            READ(OutArgs(10),*) NVert
+            ! Try to read the number of vertices as an integer
+            NVert = ProcessNumber(OutArgs(10),ErrFlag)
+            IF (ErrFlag) THEN
```

Comment:
ErrFlag is only true is OutArgs(10) is not a number, s0 4.0 and 4.1 pass through unchanged. Transition completes without crashing now, so that's good, but the non-integer values are not replaced.